src/report/repository/report-repository.service.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  ForbiddenException,
+  Injectable,
+  NotFoundException,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { HttpService } from '@nestjs/axios';
+import { ConfigService } from '@nestjs/config';
+import { catchError, map, Observable } from 'rxjs';
+
+interface ReportEntity {
+  id: string;
+  key: string;
+  value: {
+    rev: string;
+  };
+  doc: ReportDoc;
+}
```

Comment:
Generalize this and (together with FetchReportsResponse) move to general CouchDB Module? (I think I have the same interface in my draft)

```suggestion
interface CouchDbRow<T> {
  id: string;
  key: string;
  value: {
    rev: string;
  };
  doc: D;
}
```